packages/cart/src/models/line-item-adjustment.ts

```diff
@@ -1,20 +1,29 @@
 import { generateEntityId } from "@medusajs/utils"
 import {
   BeforeCreate,
+  Check,
   Entity,
   ManyToOne,
-  OnInit
+  OnInit,
+  Property,
 } from "@mikro-orm/core"
 import AdjustmentLine from "./adjustment-line"
 import LineItem from "./line-item"
 
 @Entity({ tableName: "cart_line_item_adjustment_line" })
-export default class LineItemAdjustmentLine extends AdjustmentLine {
+@Check<LineItemAdjustment>({
+  expression: (columns) => `${columns.amount} >= 0`,
+})
+export default class LineItemAdjustment extends AdjustmentLine {
   @ManyToOne(() => LineItem, {
-    joinColumn: "line_item",
-    fieldName: "line_item_id",
+    onDelete: "cascade",
+    nullable: true,
+    index: "IDX_adjustment_line_item_id",
   })
-  line_item: LineItem
+  item?: LineItem | null
+
+  @Property({ columnType: "text" })
+  item_id: string
```

Comment:
```suggestion
  @Property({ columnType: "text", nullable: true })
  item_id: string | null
```

Reply:
`item_id` is required on insert, so will let this be as is. 

`item` has to be nullable to not throw on insert. Perhaps, we should find a more appropriate pattern.

Reply:
we are also missing the convention alignment `item_id: string | null = null` @olivermrbl

Reply:
As mentioned, `item_id` can't be null, so this is not missing.

Reply:
sorry, I ve missed the commet 👍